addons/fleet/fleet.py

```diff
@@ -1,818 +1,731 @@
-        if not vehicle_id:
-            return {}
-        vehicle = self.pool.get('fleet.vehicle').browse(cr, uid, vehicle_id, context=context)
-        odometer_unit = vehicle.odometer_unit
-        driver = vehicle.driver_id.id
-        return {
-            'value': {
-                'odometer_unit': odometer_unit,
-                'purchaser_id': driver,
-            }
-        }
+    _name = 'fleet.vehicle.log.services'
+    _description = 'Services for vehicles'
+    _inherits = {'fleet.vehicle.cost': 'cost_id'}
+
+    @api.onchange('vehicle_id')
+    def on_change_vehicle(self):
+        self.odometer_unit = self.vehicle_id.odometer_unit
+        self.purchaser_id = self.vehicle_id.driver_id
 
-    def _get_default_service_type(self, cr, uid, context):
+    @api.model
+    def _get_default_service_type(self):
         try:
-            model, model_id = self.pool.get('ir.model.data').get_object_reference(cr, uid, 'fleet', 'type_service_service_8')
+            model_id = self.env.ref('fleet.type_service_service_8').id
         except ValueError:
             model_id = False
         return model_id
 
-    _inherits = {'fleet.vehicle.cost': 'cost_id'}
-    _name = 'fleet.vehicle.log.services'
-    _description = 'Services for vehicles'
-    _columns = {
-        'purchaser_id': fields.many2one('res.partner', 'Purchaser', domain="['|',('customer','=',True),('employee','=',True)]"),
-        'inv_ref': fields.char('Invoice Reference'),
-        'vendor_id': fields.many2one('res.partner', 'Supplier', domain="[('supplier','=',True)]"),
-        'cost_amount': fields.related('cost_id', 'amount', string='Amount', type='float', store=True), #we need to keep this field as a related with store=True because the graph view doesn't support (1) to address fields from inherited table and (2) fields that aren't stored in database
-        'notes': fields.text('Notes'),
-        'cost_id': fields.many2one('fleet.vehicle.cost', 'Cost', required=True, ondelete='cascade'),
-    }
+    purchaser_id = fields.Many2one('res.partner', string='Purchaser',
+                                   domain="['|', ('customer', '=', True), ('employee', '=', True)]")
+    invoice_reference = fields.Char(string='Invoice Reference')
+    vendor_id = fields.Many2one('res.partner', string='Supplier', domain="[('supplier', '=', True)]")
+    cost_amount = fields.Float(related='cost_id.amount', string='Amount', store=True)
+    # we need to keep this field as a related with store=True because the graph view doesn't support (1) to address
+    # fields from inherited table and (2) fields that aren't stored in database
+    notes = fields.Text('Notes')
+    cost_id = fields.Many2one('fleet.vehicle.cost', string='Cost', required=True, ondelete='cascade')
+
     _defaults = {
-        'date': fields.date.context_today,
+        'date': fields.Date.today(),
         'cost_subtype_id': _get_default_service_type,
         'cost_type': 'services'
     }
```

Comment:
@rco-odoo : yup. It works correctly for date, but for many2one field it is require to override field with relation.
This doesnot work : cost_subtype_id = fields.Many2one(default=_get_default_service_type)
This work : cost_subtype_id = fields.Many2one('fleet.service.type', default=_get_default_service_type)
So should i made changes accordingly?